app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,59 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :owned_post, only: [:edit, :update, :destroy]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = current_user.posts.build
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      flash[:success] = "Post created!"
+      redirect_to posts_path
+    else
+      flash[:error] = "Post unsuccessful"
+      render :new
+    end
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    @post.update(post_params)
+    flash[:success] = "Post updated!"
+    redirect_to posts_path
+  end
+
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+    flash[:notice] = 'Post deleted!'
+    redirect_to posts_path
+  end
+
+  private
+
+  def post_params
+    params[:post].permit(:caption, :image, :all_tags)
+  end
+
+  def owned_post
```

Comment:
I'm a little suspicious of this method name - normally you use a verb  unless you are expecting the method to return a thing. This is more of a "doing" method than a "returning" method, so I would expect the name to contain a verb.